### ISSUE TITLE
Cleanup CI warnings

### ISSUE THREAD

**Issue by Youssef1313** [unoplatform/uno#6670]
## What would you like to be added:

Cleaning up CI warnings. There are lots of CI warnings, from the SDK, NuGet, and the compiler.

## Why is this needed:

Just a cleanup

## For which Platform:

- [ ] iOS
- [ ] Android
- [ ] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] macOS
- [ ] Skia
  - [ ] WPF
  - [ ] GTK (Linux)
  - [ ] Tizen
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

## Anything else we need to know?

These warnings should be promoted to error after fixing.

**Comment by Youssef1313** [comment]
I fixed very few in https://github.com/unoplatform/uno/pull/6664

**Comment by retupmoc258** [comment]
I assume this is in reference to the fact that when you make anything in the XAML editor, Visual Studio doesn't recognize it?  For example, I'm getting errors for every type I use in XAML, from Page and Window, to Grid and Rectangle.  Nothing seems to be recognized by the compiler.

https://snipboard.io/FyIYfZ.jpg

**Comment by Youssef1313** [comment]
No, this issue is different. It's *mostly* fixed, except for few NuGet warnings:

![image](https://github.com/unoplatform/uno/assets/31348972/63628975-de60-4a36-b629-4cc77f2b1a67)

We should also figure out how to turn these NuGet warnings into errors after fixing them.

**Comment by jeromelaban** [comment]
These warnings will likely be removed once we move to msbuild-based nuget packaging.

**Comment by MartinZikmund** [comment]
# Repro generator

This issue was classified as not reproducible with a code sample or test.

**Reason:** CI/build enhancement to cleanup warnings.
